### PR TITLE
Append random string to cluster name to avoid conflicts b/w presubmits

### PR DESCRIPTION
**Pull request by shyamjvs**
Fixes https://github.com/kubernetes/test-infra/issues/7592

/cc @krzyzacy

### CODE REVIEW

**Comment by shyamjvs** [comment]
@rmmh What's the reason for holding this?

File: scenarios/kubernetes_e2e.py [hunk 1]
```diff
@@ -261,9 +262,11 @@ def cluster_name(cluster, build):
     """Return or select a cluster name."""
     if cluster:
         return cluster
-    if len(build) < 20:
-        return 'e2e-%s' % build
-    return 'e2e-%s' % hashlib.md5(build).hexdigest()[:10]
+    # Create a suffix based on the build number. Append a random string to it for
+    # avoiding potential conflicts across different jobs' runs (see issue #7592).
+    suffix = build if len(build) < 10 else hashlib.md5(build).hexdigest()[:10]
+    suffix += uuid.uuid4().hex[:5]
```
Comment (rmmh) [inline scenarios/kubernetes_e2e.py]: `return 'e2e-%s-%s' % (suffix, os.urandom(3).encode('hex'))` ?

Comment (shyamjvs) [reply scenarios/kubernetes_e2e.py]: Done.

Comment (BenTheElder) [reply scenarios/kubernetes_e2e.py]: why 3?

**Comment by rmmh** [comment]
I was just holding it so my comments could be addressed or ignored before it merged :)

**Pull request merged**